internal/controller/openstacknodeimagerelease_controller.go

```diff
@@ -331,27 +331,35 @@ func (r *OpenStackNodeImageReleaseReconciler) getCloudFromSecret(ctx context.Con
 	return cloud, nil
 }
 
-func findImageByName(imagesClient *gophercloud.ServiceClient, imageName string) (string, error) {
-	listOpts := images.ListOpts{
-		Name: imageName,
+func getImageID(imagesClient *gophercloud.ServiceClient, image *apiv1alpha1.CreateOpts) (string, error) {
+	var listOpts images.ListOpts
+
+	if image.ID != "" {
+		return image.ID, nil
```

Comment:
IMO we should not just return the image ID here if it is defined. 

We should check if the image with the given ID exists, and return the same ID if yes. 
If the image with the given ID does not exist we should return an empty string here, and then the image with the given ID will be crated later in code.

Reply:
It depends on what kind of behavior you want. 

Do you want to create an image if the user/creator of the cluster-stacks enters the wrong image `ID`  in the `node-images. yaml`? In this scenario he does not need to set the `URL`, does he?